pandas/core/frame.py

```diff
@@ -904,7 +909,11 @@ def from_dict(cls, data, orient='columns', dtype=None):
                     data = _from_nested_dict(data)
                 else:
                     data, index = list(data.values()), list(data.keys())
-        elif orient != 'columns':  # pragma: no cover
+        elif orient == 'columns':
+            if columns is not None:
+                raise ValueError("cannot use columns parameter with "
+                                 "orient='columns'")
```

Comment:
In principle, we could even allow it for `orient='columns'`, as `DataFrame(dict, columns=..)` will just handle this fine. 
But, not really sure of the value here.

Reply:
Could you expand on what you mean? If I pass a dict with columns to ``DataFrame()`` I get:
```
In [1]: pd.DataFrame({'A': [1, 2, 3], 'B': [4, 5, 6]}, columns = ['one', 'two'])
Out[1]:
Empty DataFrame
Columns: [one, two]
Index: []
```

Reply:
Yes, that is exactly what I meant. The `columns` keyword does a 'reindexing' operation, and does not 'overwrite' if the data has keys as well. That's the current behaviour of `DataFrame(..)`. So we could follow here the same pattern, but since this is sometimes also somewhat surprising behaviour, not sure if it is needed to have that here as well. 
So therefore my doubt :-)

Reply:
Ah yes I understand now. It makes more sense to me to raise a ValueError in the case with ``from_dict(..., orient='columns', columns=[...])`` but I can change it to not raise if we want to be consistent with ``DataFrame(dict, columns=[..])``